shared/metrics/src/main/java/io/pravega/shared/metrics/DynamicLoggerImpl.java

```diff
@@ -95,6 +95,24 @@ public Counter call() throws Exception {
     }
 
     @Override
+    public long getCounterValue(String name) {
+        Exceptions.checkNotNullOrEmpty(name, "name");
+        String counterName = name + ".Counter";
+        try {
+            Counter counter = countersCache.get(counterName, new Callable<Counter>() {
+                @Override
+                public Counter call() throws Exception {
+                    return underlying.createCounter(counterName);
```

Comment:
yeah, exactly. we need to define the behavior we want before we decide on the implementation. @shiveshr is right that a cache will often load an existing value in the case the value isn't in the case, so the logic is:

```
get (Key k):
   if( k in the cache ) then
        return value of k
   else
        load (k, v)
        return value of k
```

This is a get on the cache, though. A higher level get call (e.g., `getCounterValue`) that invokes the cache get might not want to do the whole set up for a counter metric if it is not there. Can we define the behavior first and then agree on the implementation?